govcd/ui_plugin_test.go

```diff
@@ -0,0 +1,269 @@
+	check.Assert(len(orgRefs), Equals, 0)
+
+	// Publishing/Unpublishing to/from a specific tenant, if available
+	if len(vcd.config.Tenants) > 0 {
+		existingOrg, err := vcd.client.GetOrgByName(vcd.config.Tenants[0].SysOrg)
+		check.Assert(err, IsNil)
+		existingOrgRefs := types.OpenApiReferences{{Name: existingOrg.Org.Name, ID: existingOrg.Org.ID}}
+
+		// Publish to the retrieved tenant
+		err = newUIPlugin.Publish(existingOrgRefs)
+		check.Assert(err, IsNil)
+
+		// Retrieving the published tenants, it should equal to the tenant above
+		orgRefs, err = newUIPlugin.GetPublishedTenants()
+		check.Assert(err, IsNil)
+		check.Assert(orgRefs, NotNil)
+		check.Assert(len(orgRefs), Equals, 1)
+		check.Assert(orgRefs[0].Name, Equals, existingOrg.Org.Name)
+
+		// Unpublishing from the same specific tenant
+		err = newUIPlugin.Unpublish(existingOrgRefs)
+		check.Assert(err, IsNil)
+
+		// Retrieving the published tenants, it should equal to 0 as we just unpublished it
+		orgRefs, err = newUIPlugin.GetPublishedTenants()
+		check.Assert(err, IsNil)
+		check.Assert(orgRefs, NotNil)
+		check.Assert(len(orgRefs), Equals, 0)
+	}
+
+	// Check that the plugin can be disabled and its scope changed
+	err = newUIPlugin.Update(false, false, false)
+	check.Assert(err, IsNil)
+	check.Assert(newUIPlugin.UIPluginMetadata.Enabled, Equals, false)
+	check.Assert(newUIPlugin.UIPluginMetadata.ProviderScoped, Equals, false)
+	check.Assert(newUIPlugin.UIPluginMetadata.TenantScoped, Equals, false)
+
+	// Check that the plugin can be enabled again and its scope changed
+	err = newUIPlugin.Update(true, true, true)
+	check.Assert(err, IsNil)
+	check.Assert(newUIPlugin.UIPluginMetadata.Enabled, Equals, true)
+	check.Assert(newUIPlugin.UIPluginMetadata.ProviderScoped, Equals, true)
+	check.Assert(newUIPlugin.UIPluginMetadata.TenantScoped, Equals, true)
+
+	check.Assert(newUIPlugin.IsTheSameAs(retrievedUIPlugin), Equals, true)
+	check.Assert(newUIPlugin.IsTheSameAs(&UIPlugin{UIPluginMetadata: &types.UIPluginMetadata{Vendor: "foo", Version: "1.2.3", PluginName: "bar"}}), Equals, false)
+
+	// Delete the created plugin
+	err = newUIPlugin.Delete()
+	check.Assert(err, IsNil)
+	check.Assert(*newUIPlugin.UIPluginMetadata, DeepEquals, types.UIPluginMetadata{})
+
+	// Check that the plugin was correctly deleted
+	_, err = vcd.client.GetUIPlugin(retrievedUIPlugin.UIPluginMetadata.Vendor, retrievedUIPlugin.UIPluginMetadata.PluginName, retrievedUIPlugin.UIPluginMetadata.Version)
+	check.Assert(err, NotNil)
+	check.Assert(true, Equals, strings.Contains(err.Error(), ErrorEntityNotFound.Error()))
+	_, err = vcd.client.GetUIPluginById(retrievedUIPlugin.UIPluginMetadata.ID)
+	check.Assert(err, NotNil)
+	check.Assert(true, Equals, strings.Contains(err.Error(), ErrorEntityNotFound.Error()))
+}
+
+// Test_getPluginMetadata tests that getPluginMetadata can retrieve correctly the UI plugin metadata information
+// stored inside the ZIP file.
+func Test_getPluginMetadata(t *testing.T) {
```

Comment:
Should the following tests be places in a `ui_plugin_unit_test.go` and have only tag `unit`. That way they would even be executed by Github actions. You can find similar examples in other `*_unit_test.go` files.